python_modules/dagster/dagster/_core/pipes/client.py

```diff
@@ -52,11 +52,27 @@ def run(
 
 
 class PipesClientCompletedInvocation:
+    """A wrapper for the results of a pipes client invocation, typically returned from `PipesClient.run`.
+
+    Args:
+        session (PipesSession): The Pipes session that was used to run the external process.
+        metadata (Optional[Mapping[str, Any]]): Arbitrary metadata attached to the invocation,
+            such as an external job_id or other information available on the client side. Will be injected
```

Comment:
"Arbitrary metadata that will be attached to all asset results (e.g. materializations) generated by the invocation. Useful for attaching information to asset results that is available via the external process launch API but not in the external process itself (e.g. a `job_id` param returned by the launch API call)."

Reply:
I'm always impressed by your docstrings!